src/dex-ui/hooks/hederaTokenService/useFetchTransactionDetails.tsx

```diff
@@ -0,0 +1,29 @@
+import { useQuery } from "react-query";
+import { DexService, MirrorNodeTransaction } from "../../services";
+import { HTSQueries } from "./types";
+import { isNil } from "ramda";
+import { addDelay } from "@utils";
+
+type UseTransactionDetailsKey = [HTSQueries.TransactionDetails, string];
+
+export function useFetchTransactionDetails(transactionId: string) {
+  return useQuery<MirrorNodeTransaction[] | undefined, Error, MirrorNodeTransaction[], UseTransactionDetailsKey>(
+    [HTSQueries.TransactionDetails, transactionId],
+    async () => {
+      if (isNil(transactionId) || transactionId.length === 0) return;
+      const stringArray = transactionId.split("@");
+      const idPart = stringArray[1];
+      const formattedId = idPart.replace(/[.]/g, "-");
+      const formattedTransactionId = `${stringArray[0]}-${formattedId}`;
+      /** 2 seconds addDelay is required here just immediately when we hit the API
+       * after creating the token the APi fails with 404 error
+       * */
+      await addDelay(2000);
```

Comment:
I think we should decouple the delay from this hook. The delay is not related to fetching transaction details but a side-effect of another piece of the system. Usage of this hook in another part of the app will be subjected to the delay even though it may not be relevant in that scenario. 

The ideal solution would be to subscribe and wait for a solidity event to confirm that the token was created before calling the query function. However, we don't have a system in place for event subscriptions yet. 

You could potentially place the `addDelay` call after the `createToken.mutate` (I'm not sure if this works), add an optional "delay" parameter to the `useFetchTransactionDetails` hook, or see if there is a more elegant solution using React Query. If none of these approaches work, I suggest leaving the code as is and adding more context to the description above the addDelay call.